spec/features/number_fields_spec.rb

```diff
@@ -4,9 +4,9 @@
 INVALID_INTEGER_VALUES = ["two", "2 00", "2.37"].freeze
 INVALID_MONEY_VALUES = ["two", "2 00", "$54"].freeze
 VALID_MONEY_VALUES = [
-  ["2,000", "2,000"],
+  ["2,000", "2,000", "2,000 "],
```

Comment:
What does adding this value do? I can't see where the test actually uses the third item in these. My understanding is that the first is an input value and the second is how it is displayed when you return to the form.

Additionally, because it tests what's loaded back into the form when you revisit the page, this is the result of first using `cast` to serialize the data and then `decimal_as_money_string` to deserialize it. Is there a way to test what's on the model directly, rather than once it's been through the display method? It's not vital here, I suppose, as display isn't putting trailing spaces back, but I think this would be better as a unit test, or at least closer to one.

Reply:
On this second point, you could either create a unit test of a `Form` object with a `gbp` attribute or, as a half-way house, you could modify these tests with an additional expectation about `session_contents`: i.e. get this test to expect that the value gets stored in the session as a decimal, as well as expecting it to be displayed back to the user as a formatted string.

Reply:
Cool yeah, either of those sound reasonable.